lib/src/builder/build_step.dart

```diff
@@ -13,6 +13,12 @@ abstract class BuildStep {
   /// The primary input for this build step.
   Asset get input;
 
+  /// Checks if an [Asset] by [id] exists as an input for this [BuildStep].
+  ///
+  /// If [trackAsDependency] is true, then [id] will be marked as a dependency
+  /// of all [expectedOutputs].
+  Future<bool> hasInput(AssetId id, {bool trackAsDependency: true});
```

Comment:
After some offline discussions I am going to remove this as part of the public api, at least for now, as well as removing the `addDependency` function. These combined would allow you to write a builder that cheats a bit in terms of marking its dependencies, which specifically when doing resolution would be useful. We may consider adding it back in the future, but for now it doesn't need to be there.